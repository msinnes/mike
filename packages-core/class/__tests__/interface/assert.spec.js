const BaseInterface = require('../../src/interface/base');

const assertInterface = require('../../src/interface/assert');

describe('assertInterface', () => {
  it('should be a function', () => {
    expect(assertInterface).toBeDefined();
    expect(assertInterface).toBeInstanceOf(Function);
  });

  it('should not throw an error if the input is an interface', () => {
    expect(() => {
      assertInterface(BaseInterface);
    }).not.toThrow();
  });

  it('should throw an error if input is not an interface', () => {
    expect(() => {
      assertInterface(undefined, 'message');
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      assertInterface(null, 'message');
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      assertInterface('', 'message');
    }).toThrowErrorMatchingSnapshot();
  });
});
