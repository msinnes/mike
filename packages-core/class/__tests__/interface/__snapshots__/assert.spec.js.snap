// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assertInterface should throw an error if input is not an interface 1`] = `"ClassError: message"`;

exports[`assertInterface should throw an error if input is not an interface 2`] = `"ClassError: message"`;

exports[`assertInterface should throw an error if input is not an interface 3`] = `"ClassError: message"`;
